tests/http/federation/test_matrix_federation_agent.py

```diff
@@ -1449,6 +1677,43 @@ def test_srv_fallbacks(self) -> None:
         self.reactor.pump((0.1,))
         self.successResultOf(test_d)
 
+    def test_srv_no_fallback_to_legacy(self) -> None:
```

Comment:
I was unsure about this test -- if a SRV record is found for `_matrix-fed._tcp` but is "bad" for some reason; should the look-ups continue with `_matrix._tcp`? Currently the test asserts they *do not*, but I'm unsure if that's "correct" or not.

@turt2live any thoughts here?

Reply:
The spec only falls through if the record is not found: https://spec.matrix.org/v1.8/server-server-api/#resolving-server-names

If it's an error, it should ultimately fail federation lookup I believe. I'll check what the other PRs I opened do.

Reply:
Sounds like this PR is correct then. Thanks for double checking.